### PR TITLE
[enrich] Add generic demography study

### PR DESCRIPTION
**Pull request by valeriocos**
This code introduces a generic method to execute demography studies over any index. The study requires the `author_field` param (set by default to `author_uuid`).

The study adds the min and max values to two attributes in each item: demography_min_date and demography_max_date. Note that the study is not executed in incremental mode.

### CODE REVIEW

File: grimoire_elk/enriched/enrich.py [hunk 1]
```diff
@@ -25,6 +25,7 @@
 import json
 import functools
 import logging
+import requests
```
Comment (sduenas) [inline grimoire_elk/enriched/enrich.py]: This package is not among the required packages in `requirements.txt` file or in `setup.py`.

**Review by sduenas — CHANGES_REQUESTED** [review]
I also found this error testing the code running it with `sirmordred`:

```
[2018-10-15 08:25:32,333 - grimoire_elk.elk - INFO] - Starting study: enrich_demography:git, params {}
[2018-10-15 08:25:32,333 - grimoire_elk.elk - ERROR] - Problem executing study enrich_demography:git, enrich_demography() missing 1 required positional argument: 'date_field'
Traceback (most recent call last):
  File "/home/sduenas/devel/grimoirelab/sirmordred/utils/micro.py", line 194, in <module>
    args.panels)
  File "/home/sduenas/devel/grimoirelab/sirmordred/utils/micro.py", line 62, in micro_mordred
    get_enrich(config, backend)
  File "/home/sduenas/devel/grimoirelab/sirmordred/utils/micro.py", line 112, in get_enrich
    task.execute()
  File "/home/sduenas/devel/grimoirelab/sirmordred/sirmordred/task_enrich.py", line 308, in execute
    raise e
  File "/home/sduenas/devel/grimoirelab/sirmordred/sirmordred/task_enrich.py", line 306, in execute
    self.__studies()
  File "/home/sduenas/devel/grimoirelab/sirmordred/sirmordred/task_enrich.py", line 267, in __studies
    do_studies(ocean_backend, enrich_backend, studies_args)
  File "/home/sduenas/devel/grimoirelab/elk/grimoire_elk/elk.py", line 485, in do_studies
    raise e
  File "/home/sduenas/devel/grimoirelab/elk/grimoire_elk/elk.py", line 482, in do_studies
    study(ocean_backend, enrich_backend, **params)
TypeError: enrich_demography() missing 1 required positional argument: 'date_field'
```

I think that instead of overriding `enrich_demography` with `enrich_demography(self, ocean_backend, enrich_backend, date_field, author_field="author_uuid"):` it would be better to do it with `enrich_demography(self, ocean_backend, enrich_backend)` and then pass `date_field` and `author_field` definied by each backend demograpy call. For example:

```
def enrich_demography(self, ocean_backend, enrich_backend):
    git_date_field = "utc_commit"
    git_author_field = "author_uuid"
    super.enrich_demography(ocean_backend, enrich_backend, git_date_field, git_author_field)
```

**Comment by alpgarcia** [comment]
In order to build a generic panel for all possible data sources it would be nice to add an alias `demographics` when the study finish, same way we did with onion study:
https://github.com/chaoss/grimoirelab-elk/blob/master/grimoire_elk/enriched/enrich.py#L918

**Comment by sduenas** [comment]
>In order to build a generic panel for all possible data sources it would be nice to add an alias demographics when the study finish, same way we did with onion study:

@alpgarcia where does that go? in Mordred?

**Comment by valeriocos** [comment]
generally aliases of studies are handled by elk (e.g., onion

**Comment by sduenas** [comment]
Let's do that then. @valeriocos could you add `demography` alias as @alpgarcia says?

**Review by sduenas — CHANGES_REQUESTED** [review]
I've been testing the new patch. The only problem that I found is that the alias for gerrit is not created, making unable to visualize the data with the new demographics panel.

**Review by sduenas — APPROVED** [review]
I made a mistake and I forgot to add the studies section for `gerrit` analysis. We can merge this PR.

**Pull request merged**